### PR TITLE
set cable length to 1m for backplane ports

### PR DESCRIPTION
**Pull request by zhixzhu**
Signed-off-by: Zhixin Zhu <zhixzhu@cisco.com>

#### Why I did it
backplane ports cable length need to be specified.

#### How I did it
separated handling for the specific port name.

#### How to verify it
Used "sonic-cfggen" to generate buffer config, and verified cable length of backplane ports is 1m, other ports default length.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
--> This is required for T2.
- [ ] 202211

### CODE REVIEW

**Comment by zhixzhu** [comment]
@lguohan There is 1 workflow awaiting approval, could you help approve it?

**Comment by zhixzhu** [comment]
@lguohan There is 1 workflow awaiting approval, could you help approve it?
The failure happened when running script test_pretest.py, changing the cable length of backplane ports should not have any impact on it, could you help rerun the test? I don't have privilege to do rerun.

**Comment by zhixzhu** [comment]
@yxieca @lguohan There is 1 workflow awaiting approval, could you help approve it? And rerun the pipeline?

**Comment by zhixzhu** [comment]
The failure is rsyslogd restart error:
"cmd": "docker exec -i teamd0 supervisorctl restart rsyslogd", "rc": 7, "stderr": "", "stderr_lines": [], "stdout": "rsyslogd: ERROR (not running)\nrsyslogd: ERROR (spawn error)\n"
Change cable length should not have any impact on it.

@gechiang Could you help run the pipeline again?

**Comment by lguohan** [comment]
why no unit test added for this case?

**Comment by lguohan** [comment]
@zhixzhu

**Comment by zhixzhu** [comment]
> why no unit test added for this case?

@lguohan To render buffer template, it requires minigraph, device/vendor/platform/hwsku/, there is no such directory and files for vendor cisco, and the change here is for cisco specific backplane interface name "Ethernet-BPxxx", so can't add UT for it.
cisco hwsku files are in seperate repo platform-cisco-8000 which is not open sourced.

File: files/build_templates/buffers_config.j2 [hunk 1]
```diff
@@ -93,7 +94,14 @@ def
                 {{ default_cable }}
             {%- endif %}
         {%- else -%}
-            {{ default_cable }}
+            {%- if port_name.startswith('Ethernet-BP') %}
+                {%- if 'backplane' not in ports2cable %}
+                    {%- set _ = ports2cable.update({'backplane': '1m'}) %}
+                {%- endif -%}
+                {{ ports2cable['backplane'] }}
```
Comment (abdosi) [inline files/build_templates/buffers_config.j2:101]: we already have convention of `internal` ports define for multi-asic switch . Let's reuse it and not define another notion of `backplane`. We can resuse below logic .

  ```
  {%- set roles1 = 'internal' %}
                         {%- if 'internal' not in ports2cable %}
                             {%- set _ = ports2cable.update({'internal': '5m'}) %}
                         {%- endif -%}
```

Comment (abdosi) [reply files/build_templates/buffers_config.j2:101]: Previously we were using 5m for internal links. Do we need to use 1m  ?

**Review by abdosi — CHANGES_REQUESTED** [review]

File: files/build_templates/buffers_config.j2 (hunk 1 above)
Comment (zhixzhu) [reply files/build_templates/buffers_config.j2]: I set the default cable length of internal links to 5m.
The cable length can be overwritten in vendor buffer config template.

**Comment by zhixzhu** [comment]
@abdosi @lguohan I have updated the default cable length of internal links to 5m and added UT. Please help review again, thanks.

**Review by abdosi — APPROVED** [review]

**Comment by abdosi** [comment]
LGTM.

**Comment by abdosi** [comment]
@rlhui please help merge this.

**Pull request merged**